### ISSUE TITLE
Crash on Android java.lang.AbstractStringBuilder.enlargeBuffer

### ISSUE THREAD

**Issue by chgsilva** [localytics/react-native-template-app#19]
I started to recieve this crashes on crashlytics 

react native version: 0.65.1
 localytics-react-native: "^2.8.2",

```
Fatal Exception: java.lang.OutOfMemoryError
       at java.lang.AbstractStringBuilder.enlargeBuffer(AbstractStringBuilder.java:94)
       at java.lang.AbstractStringBuilder.append0(AbstractStringBuilder.java:132)
       at java.lang.StringBuilder.append(StringBuilder.java:124)
       at android.util.Log.v(Log.java:143)
       at com.localytics.android.Logger$LogLevel.log(Logger.java:79)
       at com.localytics.android.Logger$LogLevel.access$000(Logger.java:52)
       at com.localytics.android.Logger.log(Logger.java:112)
       at com.localytics.android.Logger.log(Logger.java:108)
       at com.localytics.android.BaseProvider.query(BaseProvider.java:254)
       at com.localytics.android.AnalyticsHandler._getDataToUpload(AnalyticsHandler.java:953)
       at com.localytics.android.AnalyticsHandler._getUploadThread(AnalyticsHandler.java:977)
       at com.localytics.android.BaseHandler._upload(BaseHandler.java:380)
       at com.localytics.android.BaseHandler._upload(BaseHandler.java:341)
       at com.localytics.android.BaseHandler$2.run(BaseHandler.java:139)
       at com.localytics.android.BaseProvider.runBatchTransaction(BaseProvider.java:412)
       at com.localytics.android.BaseHandler._runBatchTransactionOnProvider(BaseHandler.java:196)
       at com.localytics.android.BaseHandler.handleMessage(BaseHandler.java:135)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

any thought on what could it cause? 

Regards

**Comment by Mriegel** [comment]
@chgsilva I believe I have seen this before. It is the result of an in memory variable for our live monitoring expanding beyond it's expected limits. I have found that this typically occurs on older devices (Android 4.4 for example). 

For the time being there is a workaround. In your `localytics.xml` file set the following:
```
<!--Optional opt out from live device logging-->
<bool name="ll_live_logging_enabled">false</bool>
```
This will disable the Localytics Live Monitoring feature which is the root cause of this problem. Obviously, this means your app will not be able to use Live Monitoring.

In the long term, this will be fixed. I have already put in a fix for SDK v6.0 which should be avaiilable for native SDKs by end of week, with wrappers coming shortly afterwards.

**Comment by chgsilva** [comment]
Hello @Mriegel ,
thanks for the quick answer. I will try the workaround for now and will look forward to the new version of the react-native sdk.

Stay safe
